test/jdk/javax/xml/jaxp/datatype/8033980/GregorianCalAndDurSerDataUtil.java

```diff
@@ -0,0 +1,133 @@
+        Duration duration = dtf.newDuration(EXPECTED_DURATION);
+        String copyRightStr = GregorianCalAndDurSerDataTemplate.ORACLE_COPY_RIGHT;
+        String classStr = GregorianCalAndDurSerDataTemplate.GREGO_CAL_DUR_SER_CLASS;
+        try(ByteArrayOutputStream baos = new ByteArrayOutputStream(); ObjectOutputStream oos = new ObjectOutputStream(baos);
+            ByteArrayOutputStream baos2 = new ByteArrayOutputStream(); ObjectOutputStream oos2 = new ObjectOutputStream(baos2)) {
+            //Serialize the given xmlGregorianCalendar
+            oos.writeObject(xmlGregorianCalendar);
+            //Serialize the given xml Duration
+            oos2.writeObject(duration);
+            Files.deleteIfExists(Path.of(testsrc,srcFilePrefix+"GregorianCalendarAndDurationSerData.java"));
+
+            copyRightStr = String.format(copyRightStr, LocalDate.now().getYear(), JDK);
+            classStr = String.format(classStr, srcFilePrefix, generatePseudoCodeForGregCalSerBytes(baos),
+                    generatePseudoCodeForDurationSerBytes(baos2));
+            String srcStr = copyRightStr + "\n" + classStr;
+            Files.writeString(Path.of(testsrc,srcFilePrefix+"GregorianCalendarAndDurationSerData.java"), srcStr);
+        }
+    }
+
+    /**
+     * Verify that Java source file is created.
+     */
+    @Test
+    void testFileCreated() {
+        assertTrue(Files.exists(Path.of(testsrc,srcFilePrefix+"GregorianCalendarAndDurationSerData.java")));
+    }
+
+    /**
+     * Generates the Java Pseudo code for serialized Gregorian Calendar byte array.
+     * @param baos Serialized GregorianCalendar ByteArrayOutputStream.
+     * @return pseudocode String for serialized Gregorian Calendar byte array.
+     */
+    public static String generatePseudoCodeForGregCalSerBytes(ByteArrayOutputStream baos) {
+        byte [] bytes = baos.toByteArray();
+        StringBuilder sb = new StringBuilder(bytes.length * 5);
+        sb.append("private final byte[] gregorianCalendarBytes = {");
+        return generatePseudoCode(sb, bytes);
+    }
+
+    /**
+     * Generates the Java Pseudo code for serialized Duration byte array.
+     * @param baos Serialized Duration ByteArrayOutputStream.
+     * @return pseudocode String for serialized Duration byte array.
+     */
+    public static String generatePseudoCodeForDurationSerBytes(ByteArrayOutputStream baos) {
+        byte [] bytesdur = baos.toByteArray();
+        StringBuilder sb = new StringBuilder(bytesdur.length * 5);
+        sb.append("private final byte[] durationBytes = {");
+        return generatePseudoCode(sb, bytesdur);
+    }
+
+    private static String generatePseudoCode(StringBuilder sb, byte [] bytes) {
+        final int linelen = 8;
+        HexFormat hex = HexFormat.of().withPrefix(" (byte) 0x").withSuffix(",");
+        for (int i = 0; i < bytes.length; i += linelen) {
+            sb.append("\n");
```

Comment:
Add some indentation before each line so the resulting code is properly indented.

Reply:
Added the indentation, reverted back to the use of Formatter class instead of HexFormat class.